client/src/api/v2/camera.js

```diff
@@ -126,33 +117,10 @@ function parseRecordingPayloadData(data) {
  * Initiate receiving status payloads
  *
  * @param {string} subComponent The sub component to init for (e.g. recording, video feed)
+ * @param {string} device Device (primary or secondary)
  */
-function initStatus(subComponent) {
-  emit(`status-camera-${subComponent}`);
-}
-
-/**
- * Returns the last status payload published
- *
- * @param {string} device Device
- * @returns {CameraRecordingStatusPayload} Payload
- */
-export function useCameraRecordingStatus(device) {
-  // Only run init once per render
-  useEffect(() => {
-    initStatus("recording");
-  }, []);
-
-  const [lastPayload, setLastPayload] = useState(null);
-
-  const update = useCallback((newPayload) => {
-    // Update last payload
-    setLastPayload(newPayload);
-  }, []);
-
-  useChannel(`status-camera-recording`, update);
-
-  return parseRecordingPayloadData(lastPayload && lastPayload[device]);
+function initStatus(subComponent, device) {
+  emit(`get-status-payload`, ["camera", subComponent].concat(device ? [device] : []));
```

Comment:
Since this is going to be used for more than just camera stuff in the future - should it be made more generic and moved to another/a new file?

Reply:
I think it's fine to leave it here for now. I think of it like `initStatus` for `camera` (thus hardcoding the "camera" part as the component). The `status.js` API currently is for the sensors so it would be a bit awkward to fit it in. Refactoring it to be more general should be quite easy anyway.